WcaOnRails/spec/controllers/teams_controller_spec.rb

```diff
@@ -180,10 +180,25 @@
       it 'cannot add overlapping membership periods for the same user'do
         member = FactoryGirl.create :user
         patch :update, id: team, team: { team_members_attributes: {"0" => { user_id: member.id, start_date: Date.today, end_date: Date.today+10, team_leader: false },
-                                                                   "1" => { user_id: member.id, start_date: Date.today+9, end_date: Date.today+20, team_leader: false }} }
+                                                                   "1" => { user_id: member.id, start_date: Date.today+9, end_date: Date.today+20, team_leader: false } } }
         invalid_team = assigns(:team)
         expect(invalid_team).to be_invalid
       end
+
+      it 'does not see an exception when tries to add another membership for the same user without start_date' do
+        member = FactoryGirl.create :user
+        expect do
+          patch :update, id: team, team: { team_members_attributes: { "0" => { user_id: member.id, start_date: Date.today+10, end_date: Date.today+5 },
+                                                                      "1" => { user_id: member.id, start_date: nil, end_date: Date.today+10 } } }
+        end.to_not raise_error
+      end
+
+      it 'does not see an exception when tries to add a member with end_date but without the start_date' do
+        member = FactoryGirl.create :user
+        expect do
+          patch :update, id: team, team: { team_members_attributes: { "0" => { user_id: member.id, start_date: nil, end_date: Date.today+5 } } }
```

Comment:
About these two tests...  I prefer to write tests without checking for errors or lack of errors whenever possible. In this particular case, I think this could be written as a database check after the patch. Of an error was thrown, then nothing would have been written to the database, presumably.